godel_process_planning/src/common_utils.cpp

```diff
@@ -0,0 +1,239 @@
+    return false;
+  }
+
+  return true;
+}
+
+trajectory_msgs::JointTrajectory
+godel_process_planning::toROSTrajectory(const godel_process_planning::DescartesTraj& solution,
+                                        const descartes_core::RobotModel& model)
+{
+  ros::Duration from_start(0.0);
+  std::vector<double> joint_point;
+  trajectory_msgs::JointTrajectory ros_trajectory; // result
+
+  for (std::size_t i = 0; i < solution.size(); ++i)
+  {
+    solution[i]->getNominalJointPose(std::vector<double>(), model, joint_point);
+
+    trajectory_msgs::JointTrajectoryPoint pt;
+    pt.positions = joint_point;
+    pt.velocities.resize(joint_point.size(), 0.0);
+    pt.accelerations.resize(joint_point.size(), 0.0);
+    pt.effort.resize(joint_point.size(), 0.0);
+
+    double time_step = solution[i]->getTiming().upper; // request descartes timing
+    if (time_step == 0.0) from_start += ros::Duration(0.1); // default time
+    else from_start += ros::Duration(time_step);
+
+    pt.time_from_start = from_start;
+
+    ros_trajectory.points.push_back(pt);
+  }
+
+  return ros_trajectory;
+}
+
+void godel_process_planning::fillTrajectoryHeaders(const std::vector<std::string>& joints,
+                                                    trajectory_msgs::JointTrajectory& traj)
+{
+  traj.joint_names = joints;
+  traj.header.frame_id = "world_frame";
+  traj.header.stamp = ros::Time::now();
+}
+
+std::vector<double> godel_process_planning::getCurrentJointState(const std::string& topic)
+{
+  sensor_msgs::JointStateConstPtr state = ros::topic::waitForMessage<sensor_msgs::JointState>(topic, ros::Duration(0.0));
+  if (!state) throw std::runtime_error("Joint state message capture failed");
+  return state->position;
+}
+
+
+
+godel_process_planning::DescartesTraj
+godel_process_planning::createLinearPath(const Eigen::Affine3d &start,
+                                         const Eigen::Affine3d &stop)
+{
+  PoseVector cart_path = interpolateCartesian(start, stop, 0.1); // 10cm spacing
+  DescartesTraj result;
+  for (std::size_t i = 0; i < cart_path.size(); ++i)
+  {
+//    result.push_back( boost::make_shared<descartes_trajectory::AxialSymmetricPt>(cart_path[i], M_PI/6.0, descartes_trajectory::AxialSymmetricPt::Z_AXIS) );
+    const Eigen::Affine3d& pose = cart_path[i];
+    Eigen::Vector3d rpy = pose.rotation().eulerAngles(0,1,2);
+    Eigen::Vector3d xyz = pose.translation();
+
+//    descartes_core::TimingConstraint tm (0.3);
+
+    double rx = rpy(0), ry = rpy(1), rz = rpy(2);
+    double x = xyz(0), y = xyz(1), z = xyz(2);
+    using namespace descartes_core;
+    using namespace descartes_trajectory;
+    TolerancedFrame frame (pose,
+                           ToleranceBase::zeroTolerance<PositionTolerance>(x,y,z),
+                           ToleranceBase::createSymmetric<OrientationTolerance>(rx,ry,rz,0,0,M_PI));
+    result.push_back(TrajectoryPtPtr(new CartTrajectoryPt(frame, 0.0, M_PI/12.0)));
```

Comment:
Lots of magic numbers and unclear implications of the stated constraints. Need to clarify.